community/io/src/test/java/org/neo4j/io/pagecache/impl/SingleFilePageSwapperTest.java

```diff
@@ -309,8 +310,25 @@ public void creatingSwapperForExternallyLockedFileMustThrow() throws Exception
         File wd = new File( "target/test-classes" ).getAbsoluteFile();
         pb.directory( wd );
         Process process = pb.start();
-        BufferedReader reader = new BufferedReader( new InputStreamReader( process.getInputStream() ) );
-        assertThat( reader.readLine(), is( LockThisFileProgram.LOCKED_OUTPUT ) );
+        BufferedReader stdout = new BufferedReader( new InputStreamReader( process.getInputStream() ) );
```

Comment:
not sure why plain redirection of child process is not used here but thats also not that important here

Reply:
Then we wouldn't be able to read the `locked` string that the program prints.

Reply:
Or send back the write that unblocks it.